code/datums/elements/embed.dm

```diff
@@ -0,0 +1,150 @@
+/datum/element/embed
+	element_flags = ELEMENT_DETACH
+	var/list/active_embeds = list()
+
+/datum/element/embed/New()
+	. = ..()
+	START_PROCESSING(SSdcs, src)
+
+/datum/element/embed/Attach(obj/item/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	if(!target.embedding)
+		target.embedding = getEmbeddingBehavior() // if it couldn't embed before, it can now (pending throwspeed, anyway)!
+
+	RegisterSignal(target, COMSIG_MOVABLE_IMPACT_ZONE, .proc/embed_check)
+	RegisterSignal(target, COMSIG_ITEM_EMBED_REMOVING_RIP, .proc/rip_out)
+	RegisterSignal(target, COMSIG_ITEM_EMBED_REMOVE_SURGERY, .proc/safe_remove)
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/byeee)
```

Comment:
This makes Travis sad. This signal is already listened for in this element, registered in the parent proc. Add the proc call to byeee to Detached or add the contents there.